### PR TITLE
Dev-docs: Developer Accounts Page

### PR DESCRIPTION
**Pull request by keeeto**
Moved the "Creating a Mantid Developer Account" on the wiki to "Developer Accounts" in the new Sphinx build developer documentation, as described in #22082.

**To test:**

Build the docs and have a quick read through them to make sure everything looks correct and sounds sensible.

**NB** there is a reference to `dev-docs-git-workflow`, which does not currently link. However, this page will have that label when it is added.

*Note* you can quick build just the `.html` developer docs using the `dev-docs-html` build target.

Refs #22082. 

**Release Notes** 

*Does not need to be in the release notes.*

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [ ] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)?
- [ ] Are the unit tests small and test the class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] Do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: dev-docs/source/DeveloperAccounts.rst [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+Account Creation
+----------------
+
+- Create a **Mantid Wiki** account; follow `this link <https://www.mantidproject.org/Special:RequestAccount>`_.
+- Sign up to the **Mantid Developers** email list; follow `this link <http://lists.mantidproject.org/mailman/listinfo/mantid-developers>`_.
+- Sign up for the **Mantid Slack** channel; follow `this link <https://mantid.slack.com/>`_.
+- If you don't already have one, sign up for a **Github** account; follow `this link <https://github.com/>`_.
+	+ Remember that your username should be easily identifiable.
+	+ Contact one of the "Owners" on `this list <https://github.com/orgs/mantidproject/people>`_ to add you to the developer team.
```
Comment (DanNixon) [inline dev-docs/source/DeveloperAccounts.rst]: You can use this link to prefill the "role:owner" search: https://github.com/orgs/mantidproject/people?query=role%3Aowner

File: dev-docs/source/DeveloperAccounts.rst [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+- **Wiki**
+    + Go to the `special pages index <https://www.mantidproject.org/Special:SpecialPages>`_.
+    + Select Login/Create Account.
+    + Select the Create Account link at the top of the box.
+    + Username should be first name (space) surname.
+    + User will be sent an email to verify.
+
+- **Github**
+	- Add the username provided to the mantid-developers team at `https://github.com/organizations/mantidproject/teams <https://github.com/organizations/mantidproject/teams>`_ Jenkins.
```
Comment (DanNixon) [inline dev-docs/source/DeveloperAccounts.rst]: Extra `Jenkins` at end of line.

**Review by DanNixon — CHANGES_REQUESTED** [review]
All looks good, just a couple of minor points.

File: dev-docs/source/DeveloperAccounts.rst (hunk 1 above)
Comment (keeeto) [reply dev-docs/source/DeveloperAccounts.rst]: Nice.

File: dev-docs/source/DeveloperAccounts.rst (hunk 2 above)
Comment (keeeto) [reply dev-docs/source/DeveloperAccounts.rst]: Good spot.

**Review by DanNixon — APPROVED** [review]

**Pull request merged**